test/common/router/header_formatter_speed_test.cc

```diff
@@ -0,0 +1,51 @@
+#include "source/common/router/header_parser.h"
+
+#include "test/common/stream_info/test_util.h"
+#include "test/mocks/common.h"
+#include "test/test_common/utility.h"
+
+#include "benchmark/benchmark.h"
+
+namespace Envoy {
+namespace Router {
+
+static void bmEvaluateHeaders(benchmark::State& state) {
+  auto request_header = Http::RequestHeaderMapImpl::create();
+  request_header->addCopy(Http::LowerCaseString("bar"), "a");
+  request_header->addCopy(Http::LowerCaseString("foo"), 1);
+  request_header->addCopy(Http::LowerCaseString("test1"), "to_overwrite");
+
+  MockTimeSystem time_system;
```

Comment:
I would avoid mocks for tests generally. I'm always suspicious they are slow.

Can you just use `Event::SimulatedTimeSystem time_system` instead?

Reply:
OK. Let me do that. Thanks.